lib/msal-common/src/auth/ClientInfo.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ClientAuthError } from "../error/ClientAuthError";
+import { StringUtils } from "../utils/StringUtils";
+import { ICrypto } from "../utils/crypto/ICrypto";
+
+/**
+ * @hidden
+ */
+export class ClientInfo {
+
+    private _uid: string;
+    get uid(): string {
+        return this._uid ? this._uid : "";
+    }
+
+    set uid(uid: string) {
+        this._uid = uid;
+    }
+
+    private _utid: string;
+    get utid(): string {
+        return this._utid ? this._utid : "";
+    }
+
+    set utid(utid: string) {
+        this._utid = utid;
+    }
+
+    constructor(rawClientInfo: string, crypto: ICrypto) {
+        if (!rawClientInfo || StringUtils.isEmpty(rawClientInfo)) {
+            this.uid = "";
```

Comment:
should these be set as empty strings instead of being null ?

Reply:
uid and utid are set in cache, so I think we do this in order to avoid null pointer exceptions. As I said above, it is probably a better pattern to handle it in the caller, so I will update the functionality.

Reply:
I have changed this, resolving comment.